omegatrade-app/backend/app/controller/simulation.controller.js

```diff
@@ -0,0 +1,123 @@
+            }
+        });
+    } else {
+        return res.status(501).json({ success: false, message: "invalid data" });
+    }
+}
+
+/**
+ * Function to Delete simulation
+ * 
+ * @method DELETE
+ */
+exports.deleteSimulation = async function (req, res) {
+    let sId = req.params.sId;
+    if (sId) {
+        await Simulation.deleteById(sId, function (err, data) {
+            if (err) { return res.json({ success: false, message: "something went wrong" }); }
+            if (data) { return res.status(200).json({ success: true, message: `deleted sucessfully` }); }
+        });
+    } else {
+       return res.status(501).json({ success: false, message: "Invalid data" });
+    }
+};
+
+exports.startSimulation = async function (req, res) {
+    try {
+        const body = req.body;
+        const [result] = await Company.findById(body.companyId);
+        if (result && result.length > 0) {
+            const company = result[0];
+            const interval = body.timeInterval * 1000;
+            const stock = fakeStockmarketgenerator.generateStockData(body.data).priceData;
+            const sId = await Simulation.create(company.companyId);
+            if (sId) {
+                var i = 0;
+                var intervalId = setInterval(async () => {
+
+                    // Generating RandomData to match with stock logic
+                    const randomValue = generateRandomValue(100, 110);
+                    const dayHigh = randomValue + generateRandomValue();
+                    const dayLow = randomValue - generateRandomValue();
+                    const stockData = {};
+                    stockData.companyStockId = uuidv4();
+                    stockData.companyId = body.companyId;
+                    stockData.companyShortCode = company.companyShortCode;
+                    stockData.date = Spanner.float(new Date().getTime());
+                    stockData.currentValue = spannerNumericRandVal(stock[i].price)
+                    stockData.open = spannerNumericRandVal(randomValue);
+                    stockData.dayHigh = spannerNumericRandVal(dayHigh);
+                    stockData.dayLow = spannerNumericRandVal(dayLow);
+                    stockData.close = spannerNumericRandValBetween(dayHigh, dayLow, 2);
+                    stockData.volume = spannerNumericRandValBetween(2000, 4000, 3);
+                    stockData.timestamp = 'spanner.commit_timestamp()';
+                    const simulation = await Simulation.findByCompanyId(body.companyId, sId);
+                    if (simulation && simulation[0]) {
+                        if (simulation[0].status) {
+                            await Company.createStockData(stockData);
+                        }
+                    } else {
+                        clearInterval(intervalId)
+                    }
+                    if (i === (body.data - 1)) {
```

Comment:
yes, you are correct. It will break the intervalled loop of simulations when the loop count reached the maximum number of data to insert(given from the frontend).
yes, I've changed the variable and added it with a code comment.